src/networking/VoiceUDPSocket.ts

```diff
@@ -74,16 +77,27 @@ export class VoiceUDPSocket extends EventEmitter {
 	}
 }
 
+const IPV4_REGEX = new RegExp(/^((25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])(\.(?!$)|$)){4}$/);
+
 /**
  * Parses the response from Discord to aid with local IP discovery.
  *
  * @param message - The received message
  */
-function parseLocalPacket(message: Buffer): SocketConfig {
+export function parseLocalPacket(message: Buffer): SocketConfig {
 	// todo, this function can throw
 	const packet = Buffer.from(message);
 	let ip = '';
 	for (let i = 4; i < packet.indexOf(0, i); i++) ip += String.fromCharCode(packet[i]);
 	const port = parseInt(packet.readUIntLE(packet.length - 2, 2).toString(10), 10);
+
+	if (!IPV4_REGEX.exec(ip)) {
+		throw new Error('Malformed IP address');
+	}
+
+	if (!port || isNaN(port)) {
```

Comment:
`global.isNaN` coerces the parameter to a number, then checks whether or not it's NaN, `Number.isNaN`, however, just does a comparison (`v !== v`), which skips some overhead:
```suggestion
	if (!port || Number.isNaN(port)) {
```

Reply:
By the way, is it intentional that you're discarding both 0 and NaN, instead of just NaN? The code could use some documentation in here.